### PR TITLE
Add basic access functions

### PR DESCRIPTION
**Pull request by ccorsin**

### CODE REVIEW

File: tests/KMS.test.ts [hunk 1]
```diff
@@ -30,6 +31,9 @@ import {
...
+  "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IjVVU1FrSVlULW9QMWZrcjQtNnRrciJ9.eyJnaXZlbl9uYW1lIjoiQ2VsaWEiLCJmYW1pbHlfbmFtZSI6IkNvcnNpbiIsIm5pY2tuYW1lIjoiY2VsaWEuY29yc2luIiwibmFtZSI6IkNlbGlhIENvcnNpbiIsInBpY3R1cmUiOiJodHRwczovL2xoMy5nb29nbGV1c2VyY29udGVudC5jb20vYS9BQ2c4b2NJWGtDVmxwaWM3QnFBajI4TXFzbmNydWxna2ozZXdEWlZzUnBtN2dPNng9czk2LWMiLCJsb2NhbGUiOiJmciIsInVwZGF0ZWRfYXQiOiIyMDIzLTA5LTI2VDA5OjM1OjI0Ljk0MloiLCJlbWFpbCI6ImNlbGlhLmNvcnNpbkBjb3NtaWFuLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJpc3MiOiJodHRwczovL2ttcy1jb3NtaWFuLmV1LmF1dGgwLmNvbS8iLCJhdWQiOiJLM2l4ZXV4blQ1azNEaGtLaHFoYjF6WG45RTYyRkV3SSIsImlhdCI6MTY5NTc0MDAxNiwiZXhwIjoxNjk2MTAwMDE2LCJzdWIiOiJnb29nbGUtb2F1dGgyfDExMTE0MjQ5NTk0OTMwODIyMzY4MCIsInNpZCI6IjZxNGtmamQzTWZ3aUFmeEp5a1FzUHNkLTRFNmhFTGt5Iiwibm9uY2UiOiJhVnBKVkRBMmRIZHdMbWRWVm5aSWQzSkpWVTlWVFhWamZtbDVPVXA2UVdkbVVsbHFTRXB0UkZoemFRPT0ifQ.nmGblaA85FTVGa5fDe6_Gy49HVe1tc0ueJUhdf6VOdBZQO6RMxDsvTz6yf58aGTl9La_EmEe0X0GgKxYfsk0hOzYKBlF5GY0B3_RfVJwe8LS3mbDtFJg05wk3ctbNxPPz1m9h2CrMZuzP63Nyk3XI05L85Izsh8K1cgdwsuk8smeeZwEyWTDIHdDQI2f9E6xly773KhnmrbdycJvbNbsqBmLKGXQjxhUIK7wJz2oIdZ-UYegZQIdZRidKbQmIQTFk_fmldz_S3TKcszYGtiedV4tOqPpRsAPZR6jTfMVvSioEQjAUCufgCRt9ZZNAN2Eu4R4YekPDaBrqhslcLmNwQ"
```
Comment (Manuthor) [inline tests/KMS.test.ts:35]: just wonder if you want to keep your own token for test?

File: tests/KMS.test.ts [hunk 2]
```diff
@@ -1133,3 +1149,56 @@ test(
...
+  async () => {
+    const kmsToken2 =
+      "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6IjVVU1FrSVlULW9QMWZrcjQtNnRrciJ9.eyJuaWNrbmFtZSI6ImNsaWVudDIiLCJuYW1lIjoiY2xpZW50MkBjb3NtaWFuLmNvbSIsInBpY3R1cmUiOiJodHRwczovL3MuZ3JhdmF0YXIuY29tL2F2YXRhci83MjZhNDZkMDdkZjY0Y2VjMDc4ZTQzNzUyYjg5MjFmMz9zPTQ4MCZyPXBnJmQ9aHR0cHMlM0ElMkYlMkZjZG4uYXV0aDAuY29tJTJGYXZhdGFycyUyRmNsLnBuZyIsInVwZGF0ZWRfYXQiOiIyMDIzLTA5LTI2VDEzOjAwOjE3LjIyNVoiLCJlbWFpbCI6ImNsaWVudDJAY29zbWlhbi5jb20iLCJlbWFpbF92ZXJpZmllZCI6ZmFsc2UsImlzcyI6Imh0dHBzOi8va21zLWNvc21pYW4uZXUuYXV0aDAuY29tLyIsImF1ZCI6IkszaXhldXhuVDVrM0Roa0tocWhiMXpYbjlFNjJGRXdJIiwiaWF0IjoxNjk1NzMzMjE5LCJleHAiOjE2OTYwOTMyMTksInN1YiI6ImF1dGgwfDY1MDQ2YzdlNWEwYWZmYzg3MjAxZWE1ZiIsInNpZCI6IkstTkdIVDFqRFJRMzFiY0ZqRHdpYnZlMkRHbGxlZzZqIiwibm9uY2UiOiJaMmR5V25GMVlucG1iSE5IUmtRNGNVbzRTRXRZUVdOdFlqZFNZelZmWVU1b1l6QmxjMFUwY3pCaE53PT0ifQ.QEVI0R3LvIJuI1m3Ew0LWk29uuSF5ObznWIWw9ctlnj8BqzwUmfCs7axDpTtRkMmKkTMq3Z0EzhpefTC_pIBntA739fJMfKdO2pGBVbBVkR5r86XyELbFeqJZNEFvq3WOmSgUtjLtE121NDkaMgZr2LxEiG6d3daD2wYQ9Ust_-JxvqDRD1Ted5-jc4w4YYir9k9lnrUZIxYrUiEKKnNWkwWZP2JHnrzB_zsgwcE4K1Hii2Q2QBT_LiOawgOnf_tbX59M-3qvQXyEfAJ-iWrHKIvDP3qkf4BUVqGkiotbsCMMntyCfgwm35Wzmo2nunw72i-4bkQjh9Cz9GpikFWDA"
```
Comment (Manuthor) [inline tests/KMS.test.ts:1157]: add a small comment indicating the origin of this token (at least the user email)

File: tests/KMS.test.ts [hunk 3]
```diff
@@ -1133,3 +1149,56 @@ test(
...
+
+    const client = new KmsClient(
+      `http://${process.env.KMS_HOST || "localhost"}:9998`,
+      kmsToken,
+    )
+    if (!(await client.up())) {
+      console.error("No KMIP server. Skipping test")
+    }
```
Comment (Manuthor) [inline tests/KMS.test.ts:1165]:
```suggestion
```

IMHO we should fail all the KMS tests where the KMS server is required and not up.

File: src/kms/kms.ts [hunk 4]
```diff
@@ -919,9 +919,131 @@ export class KmsClient {
...
   }
+
+  /**
+   * Grant access to a KmsObject for a specific user
+   * @param uniqueIdentifier the unique identifier of the object to import
+   * @param userIdentifier the unique identifier of the user to grant access to
+   * @param operationType KMIP operation type to grant access for
+   * @returns response from KMS server
+   */
+  public async grantAccess(
```
Comment (Manuthor) [inline src/kms/kms.ts:930]: LGTM. 
Just a comment: 
- `grandAccess` and `revokeAccess` are very similar
- same for `listAccess`, `listOwnedObjects` and `listObtainedObjects`

Also, should we return an object where the JSON response is parsed? (I guess not but just wondering)

Comment (ccorsin) [reply src/kms/kms.ts:930]: I think it's more intuitive to keep seperated functions - and not passe the action as a parameter - but we can do differently if you want.
Same for the result - as you wish :)

File: tests/KMS.test.ts (hunk 3 above)
Comment (ccorsin) [reply tests/KMS.test.ts]: Yes - I put the check in a beforeAll function to setup the client. So the check is only done once

**Review by Manuthor — APPROVED** [review]

**Pull request merged**